presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -517,22 +521,35 @@ else if (coordinatorSidecar) {
         return server;
     }
 
-    private boolean allNodesGloballyVisible()
+    private void waitForAllNodesGloballyVisible(long startTimeInMs)
+            throws Exception
     {
-        int expectedActiveNodesForRm = externalWorkers.size() + servers.size();
-        int expectedActiveNodesForCoordinator = externalWorkers.size() + servers.size();
+        int expectedActiveNodes = externalWorkers.size() + servers.size();
+        Duration timeout = new Duration(100, SECONDS);
 
-        for (TestingPrestoServer server : servers) {
-            AllNodes allNodes = server.refreshNodes();
-            int activeNodeCount = allNodes.getActiveNodes().size();
+        while (true) {
+            for (TestingPrestoServer server : servers) {
+                AllNodes allNodes = server.refreshNodes();
+                int activeNodeCount = allNodes.getActiveNodes().size();
 
-            if (!allNodes.getInactiveNodes().isEmpty() ||
-                    (server.isCoordinator() && activeNodeCount != expectedActiveNodesForCoordinator) ||
-                    (server.isResourceManager() && activeNodeCount != expectedActiveNodesForRm)) {
-                return false;
+                if (!allNodes.getInactiveNodes().isEmpty()) {
+                    if (nanosSince(startTimeInMs).compareTo(timeout) >= 0) {
```

Comment:
Nit : Abstract the checked condition with custom exception message as a lambda maybe ?

Reply:
Done